### ISSUE TITLE
panic: send on closed channel

### ISSUE THREAD

**Issue by xeoncross** [Xeoncross/goworkqueue#2]
Encountering a panic after the `queue.done` channel has already been closed. Code outside the workers is `queue.Add()` work after an `os.Signal` event has triggering a `queue.Close()`.

```
close(q.done)
Add = true
Add = false
close(q.Jobs)
Add
panic: send on closed channel
```

**Comment by ghostiam** [comment]
check `queue.Closed()` before add

**Comment by xeoncross** [comment]
Well, it's not that simple as [both Closed() and Add() already check the `q.done` channel](https://github.com/Xeoncross/goworkqueue/blob/cc56da01207926a0192f9fde4b4911e5b409bab6/goworkqueue.go#L80-L104).

**Comment by xeoncross** [comment]
Resolved in commit a603a0fc422adbb78b9c4be302d991b5f5da381d

**Comment by ghostiam** [comment]
Hello. Thanks for the project, it helps a lot.

> There seems to be a theoretical chance of a race condition by Add()
https://github.com/Xeoncross/goworkqueue/blob/master/queue.go#L62

As it turns out, this is not a theoretical chance...
I copied your library into my project and added support for generics.
For the second day in a row, GitLab CI has been unable to pass the pre-build tests because it receives a "panic: send on closed channel" error.

```
panic: send on closed channel

goroutine 207 [running]:
<softname>_v2/pkg/goworkqueue.(*Queue[...]).Add(0xc00007c390?, 0x907)
	/builds/<softname>/<softname>_v2/pkg/goworkqueue/queue.go:103 +0x8f
<softname>_v2/pkg/goworkqueue.TestQueue.func3(0x0?)
	/builds/<softname>/<softname>_v2/pkg/goworkqueue/queue_test.go:53 +0x85
created by <softname>_v2/pkg/goworkqueue.TestQueue
	/builds/<softname>/<softname>_v2/pkg/goworkqueue/queue_test.go:48 +0xb7
FAIL	<softname>_v2/pkg/goworkqueue	0.024s
```

In real code, I don't interrupt the work of the workers, so I never got such an error.
